openassessment/data.py

```diff
@@ -498,6 +500,102 @@ def collect_ora2_data(cls, course_id):
         ]
         return header, rows
 
+    @classmethod
+    def collect_ora2_summary(cls, course_id):
+        """
+        Query database for aggregated ora2 summary data.
+
+        Args:
+            course_id (string) - the course id of the course whose data we would like to return
+
+        Returns:
+            A tuple containing two lists: headers and data.
+
+            headers is a list containing strings corresponding to the column headers of the data.
+            data is a list of lists, where each sub-list corresponds to a row in the table of all the data
+                for this course.
+
+        """
+
+        items = AssessmentWorkflow.objects.filter(course_id=course_id)
+
+        # need the workflow steps set and sorted here so the data columns line
+        # up with the headers
+        steps = sorted(AssessmentWorkflow.STEPS)
+
+        rows = []
+        for aw in items:
+            statuses = aw.status_details()
+            submission_dict = sub_api.get_submission_and_student(aw.submission_uuid)
+
+            steps_statuses = []
+            peers_graded = 0
+            graded_by_count = 0
+            for step in steps:
+                if not statuses.get(step):
+                    steps_statuses.append(_('NA'))
+                    steps_statuses.append(_('NA'))
+                    continue
+
+                if statuses[step]['complete']:
+                    steps_statuses.append(_('complete'))
+                else:
+                    steps_statuses.append(u'')
+
+                if statuses[step]['graded']:
+                    steps_statuses.append(_('complete'))
+                else:
+                    steps_statuses.append(u'')
+
+                if step == 'peer':
+                    peers_graded = statuses[step]['peers_graded_count']
+                    graded_by_count = statuses[step]['graded_by_count']
+
+            is_staff_grade_received = 1 if aw.staff_score_exists() else 0
+            is_final_grade_received = 1 if aw.status == AssessmentWorkflow.STATUS.done else 0
+            final_grade = aw.score if aw.score is not None else ''
```

Comment:
From testing on the sandbox (see [edX_DemoX_Demo_Course_ORA_summary_2019-07-15-1025.csv](https://ocim-pr20955-sandbox-opencraft-hosting.s3.amazonaws.com/pr20955_sandbox_opencraft_hosting/grades-download/bb6d638a296059742509a0d319ddf8456b6dbf9a/edX_DemoX_Demo_Course_ORA_summary_2019-07-15-1025.csv)), the `final_grade` column reports this full JSON data string: 
```
{'submission': 2, 'created_at': datetime.datetime(2019, 7, 15, 10, 24, 8, 940437, tzinfo=<UTC>), 'points_earned': 6, 'submission_uuid': u'09357d3d-0ba4-494d-9398-172c60c9f9ba', 'annotations': [], 'student_item': 2, 'points_possible': 19}
```

So maybe we need two columns here, `final_grade_points_earned` and `final_grade_points_possible`.

And tests updated to test for this case :)

Reply:
Nice catch!

Reply:
Ok, this is interesting - the existing tests *should* have a user with a final grade (according to the `test_collect_ora2_data` test), but the AssessmentWorkflow items return None. Can these be out of sync somehow? Perhaps it's related to the fact that the collect_ora2_data  method directly gets the submissions information so it has a score even if the ora workflow isn't complete. Not sure what is a more correct source of information in this context...

Reply:
Hmm actually I think it's the way the existing test setUp sets the data - it's directly on the submission objects instead of involving the assessment workflow procedure. Might need to create a new test class.

Reply:
Solved.